freshmaker/handlers/koji/rebuild_images_on_rpm_advisory_change.py

```diff
@@ -423,8 +415,28 @@ def _find_images_to_rebuild(self, errata_id):
             "Going to find all the container images to rebuild as "
             "result of %r update.", affected_nvrs)
         batches = lb.find_images_to_rebuild(
-            affected_nvrs, content_sets,
+            affected_nvrs, self.event.content_sets,
             filter_fnc=self._filter_out_not_allowed_builds,
             published=published, release_categories=release_categories,
             leaf_container_images=leaf_container_images)
         return batches
+
+
+    def _set_event_content_sets(self, errata):
+        """
+        Set 'content_sets' for currently processed event
+
+        :param Errata errata: Errata api to get Pulp repos IDs
+        """
+        # Other handler could set 'content_sets' for the event
+        if not hasattr(self.event, 'content_sets'):
+            # Use the errata_id to find out Pulp repository IDs from Errata Tool
+            # and furthermore get content_sets from Pulp where signed RPM will end
+            # up eventually when advisories are shipped.
+            pulp_repo_ids = list(set(errata.get_pulp_repository_ids(errata_id)))
+
+            pulp = Pulp(server_url=conf.pulp_server_url,
+                        username=conf.pulp_username,
+                        password=conf.pulp_password)
+            self.event.content_sets = pulp.get_content_set_by_repo_ids(
```

Comment:
What is the benefit of setting it on the event?

Reply:
It's just as I've mentioned above, if one event will be handled by more handlers, it makes no sense to request same content_sets again. So just to make it more efficient I've decided to use Event to store this information.